app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadCache.kt

```diff
@@ -75,11 +75,9 @@ class DownloadCache(
 
         checkRenew()
 
-        val files = mangaFiles[manga.id]?.toSet() ?: return false
-        return files.any { file ->
-            provider.getValidChapterDirNames(chapter).any {
-                it.toLowerCase() == file.toLowerCase()
-            }
+        val files = mangaFiles[manga.id]?.toHashSet() ?: return false
+        return provider.getValidChapterDirNames(chapter).any {
+            it in files
```

Comment:
I had the lowercase check before since chapter names sometimes update with a case change and it's annoying. I'll see if it's there's another way to change the casing logic before hand but this is fine as is

Reply:
I think if you pull from main app the rename folder when source changed chapter name. It should be fine